test/integration/bastion/bastion_test.go

```diff
@@ -0,0 +1,598 @@
+
+	return &infrastructure{
+		VPCID:                 vpcID,
+		SubnetID:              subnetID,
+		WorkerSecurityGroupID: workerSecurityGroupID,
+	}
+}
+
+func teardownInfrastructure(ctx context.Context, logger *logrus.Entry, awsClient *awsclient.Client, infra *infrastructure) {
+	Expect(integration.DestroySubnet(ctx, logger, awsClient, infra.SubnetID)).To(Succeed())
+	Expect(integration.DestroySecurityGroup(ctx, logger, awsClient, infra.WorkerSecurityGroupID)).To(Succeed())
+	Expect(integration.DestroyVPC(ctx, logger, awsClient, infra.VPCID)).To(Succeed())
+}
+
+func setupShootEnvironment(ctx context.Context, logger *logrus.Entry, c client.Client, namespace *corev1.Namespace, cluster *extensionsv1alpha1.Cluster) {
+	Expect(c.Create(ctx, namespace)).To(Succeed())
+	Expect(c.Create(ctx, cluster)).To(Succeed())
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      v1beta1constants.SecretNameCloudProvider,
+			Namespace: namespace.Name,
+		},
+		Data: map[string][]byte{
+			aws.AccessKeyID:     []byte(*accessKeyID),
+			aws.SecretAccessKey: []byte(*secretAccessKey),
+		},
+	}
+	Expect(c.Create(ctx, secret)).To(Succeed())
+}
+
+func teardownShootEnvironment(ctx context.Context, logger *logrus.Entry, c client.Client, namespace *corev1.Namespace, cluster *extensionsv1alpha1.Cluster) {
+	Expect(client.IgnoreNotFound(c.Delete(ctx, namespace))).To(Succeed())
+	Expect(client.IgnoreNotFound(c.Delete(ctx, cluster))).To(Succeed())
+}
+
+func setupBastion(ctx context.Context, logger *logrus.Entry, awsClient *awsclient.Client, c client.Client, name string, cluster *controller.Cluster) (*extensionsv1alpha1.Bastion, *bastionctrl.Options) {
+	bastion, err := newBastion(name)
+	Expect(err).NotTo(HaveOccurred())
+
+	options, err := bastionctrl.DetermineOptions(ctx, bastion, cluster, awsClient)
+	Expect(err).NotTo(HaveOccurred())
+
+	Expect(c.Create(ctx, bastion)).To(Succeed())
+
+	return bastion, options
+}
+
+func teardownBastion(ctx context.Context, logger *logrus.Entry, c client.Client, bastion *extensionsv1alpha1.Bastion) {
+	By("delete bastion")
+	Expect(client.IgnoreNotFound(c.Delete(ctx, bastion))).To(Succeed())
+
+	By("wait until bastion is deleted")
+	err := extensions.WaitUntilExtensionCRDeleted(
+		ctx,
+		c,
+		logger,
+		func() extensionsv1alpha1.Object { return &extensionsv1alpha1.Bastion{} },
+		extensionsv1alpha1.BastionResource,
+		bastion.Namespace,
+		bastion.Name,
+		10*time.Second,
+		16*time.Minute,
+	)
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func newCluster(name string, amiID string) (*extensionsv1alpha1.Cluster, *controller.Cluster) {
+	providerConfig := &awsv1alpha1.CloudProfileConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "aws.provider.extensions.gardener.cloud/v1alpha1",
+			Kind:       "CloudProfileConfig",
+		},
+		MachineImages: []awsv1alpha1.MachineImages{
+			{
+				Name: "ubuntu",
+				Versions: []awsv1alpha1.MachineImageVersion{
+					{
+						Version: "20.04.20210223",
```

Comment:
can we have that version also as a constant like the image name? so that there is one place for such configuration